vault/diagnose/helpers.go

```diff
@@ -154,7 +184,7 @@ func Test(ctx context.Context, spanName string, function func(context.Context) e
 // complete within the timeout, e.g.
 //
 // diagnose.Test(ctx, "my-span", diagnose.WithTimeout(5 * time.Second, myTestFunc))
-func WithTimeout(d time.Duration, f func(context.Context) error) func(ctx context.Context) error {
+func WithTimeout(d time.Duration, f func(context.Context) error) func(context.Context) error {
```

Comment:
might be helpful to the reader to define a type for `func(context.Context) error`?

Reply:
For code cleanliness, yeah, good idea.